packages/labs/analyzer/src/lib/model.ts

```diff
@@ -428,7 +431,10 @@ export interface ClassDeclarationInit extends DeclarationInit {
 }
 
 export class ClassDeclaration extends Declaration {
-  readonly node: ts.ClassLikeDeclaration;
+  readonly node:
+    | ts.ClassDeclaration
+    | ts.ClassLikeDeclaration
+    | ts.CallExpression;
```

Comment:
fwiw im not a fan of this since its a `ClassDeclaration` we're in, but im not sure how else we'd do it (to separate `ClassDeclaration` and mixins somehow)

Reply:
Yeah, the TS `node` field on the different declaration models is really of sort of dubious value; the declaration types in the model are sort of "semantic" declarations from the point of view of the module exports, not 1:1 mappings to AST types. 

Meaning, we're taking the stance that it's more useful to represent `export const MyClassWithSomeMixin = SomeMixin(MyClass)` as a ClassDeclaration than just a VariableDeclaration with a class type, since it's semantically almost identical to `export class MyClassWithSomeMixin extends SomeMixin(MyClass) {}`; consumers of the module should just care that it's a class, not how it was assembled.

The `node`s are only really going to be useful for e.g. maybe getting linkages back to source code locations, and I think it's ok to have this weirdness.